app/decorators/sensor_decorator.rb

```diff
@@ -6,9 +6,13 @@ def initialize(sensor)
 
   def last_value(opts={})
     r = @sensor.last_reading(opts)
-    v = r.calibrated_value
     u = @sensor.sensor_type.unit
-    "#{format("%.1f", v)}#{u}"
+    unless r.nil?
```

Comment:
In what scenario do we need to check for `nil` actually? Could you write an Rspec test for that?